tasks/flush_repo.py

```diff
@@ -32,16 +35,28 @@
 log = logging.getLogger(__name__)
 
 
+@dataclass
+class FlushRepoTaskReturnType(object):
+    error: Optional[str] = None
+    deleted_commits_count: int = 0
+    delete_branches_count: int = 0
+    deleted_pulls_count: int = 0
+    deleted_archives_count: int = 0
+
+
 class FlushRepoTask(BaseCodecovTask, name="app.tasks.flush_repo.FlushRepo"):
-    async def run_async(self, db_session, *, repoid: int, **kwargs):
-        log.info("Deleting repo contents", extra=dict(repoid=repoid))
-        repo = db_session.query(Repository).filter_by(repoid=repoid).first()
+    @sentry_sdk.trace
+    def _delete_archive(self, repo: Repository) -> int:
         archive_service = ArchiveService(repo)
         deleted_archives = archive_service.delete_repo_files()
```

Comment:
Should this fn be async? I suppose we don't necessarily care to wait for the files to be deleted, but we don't know for certain if the call succeeded otherwise right?

Reply:
Yea agree, not part of this scope, just found it interesting that the archive service wasn't aysnc in the first place, makes sense